pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -459,13 +479,38 @@ void connectionOpened(final ClientCnx cnx) {
         log.info("[{}][{}] Subscribing to topic on cnx {}", topic, subscription, cnx.ctx().channel());
 
         long requestId = client.newRequestId();
-        cnx.sendRequestWithId(Commands.newSubscribe(topic, subscription, consumerId, requestId, getSubType(),
-                priorityLevel, consumerName), requestId).thenRun(() -> {
+
+        int currentSize;
+        MessageIdImpl startMessageId;
+        synchronized (this) {
+            currentSize = incomingMessages.size();
+            startMessageId = clearReceiverQueue();
```

Comment:
I don't expect this to have a practical impact on the consumer performance, since it only happens on reconnections.